### ISSUE TITLE
Code completion partially works to certain line in MXML file

### ISSUE THREAD

**Issue by rat-moonshine** [prominic/Moonshine-IDE#145]
- Create a new Flex desktop project
- Adds any arbitrary method to its `fx:Script` tag
- Try to add `creationComplete` in the line as shown in the following image (it is important)
![image](https://user-images.githubusercontent.com/24370870/35027346-3a99e2f2-fb76-11e7-8c5f-04e65ee376ed.png)
- Adding `creationComplete` through code-completion should work
- Populating the available methods inside the **\<your cursor here\>** (`creationComplete="<your cursor here>"`) position through code-completion, do not works

The above tested in Windows.

**Comment by piotrzarzycki21** [comment]
Hi @joshtynjala,
I just checked that issue and it looks like this is something on the server language. Basically if we are trying to show CC in the same line with some namespaces I'm getting empty list of items from the language server.

**Comment by joshtynjala** [comment]
Can reproduce in both Moonshine and VSCode.

Interestingly, completion works as expected on a line that contains only namespaces defined as a package, like these:

``` mxml
xmlns:local="*" xmlns:local="com.example.*"
```

I will investigate.

**Comment by piotrzarzycki21** [comment]
There is more interesting that I couldn't reproduce it in VSCode. :) I'm glad that you were able to.

**Comment by JoelProminic** [comment]
The original issue reported by @rat-moonshine still triggers for me on both Mac OS and Windows.  I can't complete "abcd()" when it is on the same line as "xmlns:mx", but it does work when I move creationComplete to a new line.

**Comment by joshtynjala** [comment]
@JoelProminic I cannot reproduce. When I press Ctrl+Space inside creationComplete="", the completion list appears and abcd() is one of the available options.

**Comment by piotrzarzycki21** [comment]
@JoelProminic could you try again with updated language server ?

**Comment by JoelProminic** [comment]
Here are my steps to reproduce (I tested with a Mac OS build):
1. Create a new Flex Desktop Project with File > New > Flex Desktop Project
2. Build & Run the project
3. Add a method called abcd() in the <fx:Script> block
4. Put your cursor at the end of the <s:WindowedApplication> tag (after minHeight, same line as xmlns:mx)
5. Type ' creationComplete="abc'
6. Press Ctrl+Space.  BUG:  Nothing happens
7. Clear the added text
8. Hit Enter to add a new line to the <s:WindowedApplication tag
9. Type 'creationComplete="abc'
10. Press Ctrl+Space.  The completion prompt appears after a noticeable delay


While testing this today, I noticed a another, potentially unrelated issue.
11. Select "abcd()" and hit Enter
12. BUG:  The inserted string was 'abcd="' instead.  So, the whole line was:  'creationComplete="abcd="">'

**Comment by piotrzarzycki21** [comment]
@JoelProminic can you try to retest that with 1.12.0. I tested it today and everything seems to be working with your scenario.

**Comment by rat-moonshine** [comment]
I tested with latest pull from Git on macOS. While in my test the 2nd bug report of @JoelProminic seems fixed, the former one remains.

**Comment by rat-moonshine** [comment]
As I've mentioned my test reports (above) in Windows, macOS tests also result similar. However, I encountered an index range error while doing this test:

11. Select "abcd()" and hit Enter
12. BUG: The inserted string was 'abcd="' instead. So, the whole line was: 'creationComplete="abcd="">'

Although, the functionality completes after the debugger error - user may not encounter this error in a packaged application, though.

```
RangeError: Error #2006: The supplied index is out of bounds.
	at flash.text.engine::TextLine/getAtomBounds()
	at actionScripts.ui.editor.text::TextLineRenderer/drawCaret()[/Users/ProminicGITBranch/ide/MoonshineSharedCore/src/actionScripts/ui/editor/text/TextLineRenderer.as:258]
	at actionScripts.ui.editor.text::TextLineRenderer/set caretPosition()[/Users/ProminicGITBranch/ide/MoonshineSharedCore/src/actionScripts/ui/editor/text/TextLineRenderer.as:127]
	at actionScripts.ui.editor.text::TextEditor/updateSelection()[/Users/ProminicGITBranch/ide/MoonshineSharedCore/src/actionScripts/ui/editor/text/TextEditor.as:1161]
	at actionScripts.plugin.core.mouse::MouseManagerPlugin/setFocusToTextEditor()[/Users/ProminicGITBranch/ide/MoonshineSharedCore/src/actionScripts/plugin/core/mouse/MouseManagerPlugin.as:87]
	at actionScripts.plugin.core.mouse::MouseManagerPlugin/onCursorUpdated()[/Users/ProminicGITBranch/ide/MoonshineSharedCore/src/actionScripts/plugin/core/mouse/MouseManagerPlugin.as:59]
	at flash.display::Stage/set focus()
	at mx.core::UIComponent/setFocus()[D:\flex_sdk\Sources\flex-sdk\frameworks\projects\framework\src\mx\core\UIComponent.as:10437]
	at mx.managers::FocusManager/setFocus()[D:\flex_sdk\Sources\flex-sdk\frameworks\projects\framework\src\mx\managers\FocusManager.as:622]
	at mx.managers::FocusManager/activate()[D:\flex_sdk\Sources\flex-sdk\frameworks\projects\framework\src\mx\managers\FocusManager.as:888]
	at mx.managers.systemClasses::ActiveWindowManager/deactivateForm()[D:\flex_sdk\Sources\flex-sdk\frameworks\projects\framework\src\mx\managers\systemClasses\ActiveWindowManager.as:267]
	at mx.managers.systemClasses::ActiveWindowManager/deactivate()[D:\flex_sdk\Sources\flex-sdk\frameworks\projects\framework\src\mx\managers\systemClasses\ActiveWindowManager.as:221]
	at mx.managers.systemClasses::ActiveWindowManager/removeFocusManager()[D:\flex_sdk\Sources\flex-sdk\frameworks\projects\framework\src\mx\managers\systemClasses\ActiveWindowManager.as:411]
	at mx.managers::PopUpManagerImpl/popupRemovedHandler()[D:\flex_sdk\Sources\flex-sdk\frameworks\projects\framework\src\mx\managers\PopUpManagerImpl.as:1245]
	at flash.display::DisplayObjectContainer/removeChild()
	at mx.managers::SystemManager/http://www.adobe.com/2006/flex/mx/internal::rawChildren_removeChild()[D:\flex_sdk\Sources\flex-sdk\frameworks\projects\framework\src\mx\managers\SystemManager.as:2152]
	at mx.managers::SystemManager/removeChild()[D:\flex_sdk\Sources\flex-sdk\frameworks\projects\framework\src\mx\managers\SystemManager.as:1795]
	at mx.managers::PopUpManagerImpl/removePopUp()[D:\flex_sdk\Sources\flex-sdk\frameworks\projects\framework\src\mx\managers\PopUpManagerImpl.as:671]
	at mx.managers::PopUpManager$/removePopUp()[D:\flex_sdk\Sources\flex-sdk\frameworks\projects\framework\src\mx\managers\PopUpManager.as:238]
	at actionScripts.ui.editor.text::CompletionManager/closeCompletionList()[/Users/ProminicGITBranch/ide/MoonshineSharedCore/src/actionScripts/ui/editor/text/CompletionManager.as:134]
	at actionScripts.ui.editor.text::CompletionManager/onMenuKey()[/Users/ProminicGITBranch/ide/MoonshineSharedCore/src/actionScripts/ui/editor/text/CompletionManager.as:266]
```

**Comment by piotrzarzycki21** [comment]
Did you build version from Flash Builder or is it version from Bamboo ?

**Comment by joshtynjala** [comment]
> Type ' creationComplete="abc'
> Press Ctrl+Space. BUG: Nothing happens

In my previous attempts to reproduce, I missed that Joel added the extra step of typing *abc* before you press Ctrl+Space. I can reproduce now.

**Comment by rat-moonshine** [comment]
As a correction to @JoelProminic 's comment, I did not able to reproduce the error on 12. 6 is still failing for me, too.

**Comment by piotrzarzycki21** [comment]
@rat-moonshine It's working for me with Josh's update to 0.12.0.

**Comment by JoelProminic** [comment]
@piotrzarzycki21 I am still seeing the bugs from steps 6 and 12 from my above procedure.  I am using the latest Bamboo build on macOS.

**Comment by piotrzarzycki21** [comment]
@JoelProminic would it be possible to get some gif recording for that one ?

**Comment by piotrzarzycki21** [comment]
@JoelProminic can you take a look again - it's been a lot of updates since your last attempts.

**Comment by JoelProminic** [comment]
I see the same behavior I reported before.  Here are the steps I used, which are the same as what I used earlier:
1. Create a new Flex Desktop Project with File > New > Flex Desktop Project
2. Build & Run the project
3. Add a method called abcd() in the <fx:Script> block
4. Put your cursor at the end of the <s:WindowedApplication> tag (after minHeight, same line as xmlns:mx)
5. Type ' creationComplete="abc'
6. Press Ctrl+Space.  BUG:  Nothing happens
7. Clear the added text
8. Hit Enter to add a new line to the <s:WindowedApplication tag
9. Type 'creationComplete="abc'
10. Press Ctrl+Space.  The completion prompt appears after a noticeable delay
11. Select "abcd()" and hit Enter
12. BUG:  The inserted string was 'abcd="' instead.  So, the whole line was:  'creationComplete="abcd="">'

**Comment by joshtynjala** [comment]
This issue will be fixed in the upcoming vscode-as3mxml v0.22.

**Comment by joshtynjala** [comment]
Since I've updated to v0.23.2, this issue should now be resolved.
